extensions/websockets-next/runtime/src/main/java/io/quarkus/websockets/next/HttpUpgradeCheck.java

```diff
@@ -45,6 +46,19 @@ default boolean appliesTo(String endpointId) {
 
     interface HttpUpgradeContext {
 
+        /**
+         * @return userData {@link UserData}; mutable user data to associate with an upgraded connection
+         * @see WebSocketConnection#userData()
+         */
+        UserData userData();
+
+        /**
+         * @return pathParams {@link Map Map&lt;String, String&gt;}; a map of named parameters as defined in path
+         *         declaration with their actual values
+         * @see RoutingContext#pathParams()
+         */
+        Map<String, String> pathParams();
```

Comment:
I wonder if we should follow the `Connection` API  and define `String pathParam(String name)` instead?

Reply:
> I wonder if we should follow the `Connection` API and define `String pathParam(String name)` instead?

+1

Reply:
I like it. I'll make the change.